scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/package.scala

```diff
@@ -0,0 +1,127 @@
+import org.apache.parquet.hadoop.ParquetInputFormat
+
+import scala.reflect.ClassTag
+
+/**
+ * Main package for Parquet Avro APIs. Import all.
+ *
+ * {{{
+ * import com.spotify.scio.parquet.avro._
+ * }}}
+ */
+package object avro {
+
+  /** Alias for [[me.lyh.parquet.avro.Projection]]. */
+  val Projection = me.lyh.parquet.avro.Projection
+
+  /** Alias for [[me.lyh.parquet.avro.Predicate]]. */
+  val Predicate = me.lyh.parquet.avro.Predicate
+
+  /** Enhanced version of [[ScioContext]] with Parquet Avro methods. */
+  implicit class ParquetAvroScioContext(val self: ScioContext) extends AnyVal {
+
+    /**
+     * Get an SCollection for a Parquet file as Avro records.
+     *
+     * Note that due to limitations of the underlying [[HadoopInputFormatIO]],
+     * Avro [[org.apache.avro.generic.GenericRecord GenericRecord]] and dynamic work rebalancing
+     * are not supported. Without the latter, pipelines may not autoscale up or down during the
+     * initial read and subsequent fused transforms.
+     *
+     * @group input
+     */
+    def parquetAvroFile[T <: SpecificRecordBase : ClassTag](path: String,
+                                                            projection: Schema = null,
+                                                            predicate: FilterPredicate = null)
+    : SCollection[T] = self.requireNotClosed {
+      if (self.isTest) {
+        self.getTestInput(AvroIO[T](path))
+      } else {
+        val cls = ScioUtil.classOf[T]
+        val readSchema = cls.getMethod("getClassSchema").invoke(null).asInstanceOf[Schema]
+
+        val job = Job.getInstance()
+        setInputPaths(job, path)
+        job.setInputFormatClass(classOf[AvroParquetInputFormat[T]])
+        job.getConfiguration.setClass("key.class", classOf[Void], classOf[Void])
+        job.getConfiguration.setClass("value.class", cls, cls)
+
+        AvroParquetInputFormat.setAvroReadSchema(job, readSchema)
+        if (projection != null) {
+          AvroParquetInputFormat.setRequestedProjection(job, projection)
+        }
+        if (predicate != null) {
+          ParquetInputFormat.setFilterPredicate(job.getConfiguration, predicate)
+        }
+
+        val source = HadoopInputFormatIO.read[JBoolean, T]()
+          .withConfiguration(job.getConfiguration)
+          .withKeyTranslation(new SimpleFunction[Void, JBoolean]() {
+            override def apply(input: Void): JBoolean = true // workaround for NPE
+          })
+        self
+          .wrap(self.applyInternal(source))
+          .map(_.getValue)
+      }
+    }
+
+    private def setInputPaths(job: Job, path: String): Unit = {
+      // These are needed since `FileInputFormat.setInputPaths` validates paths locally and
+      // requires the user's GCP credentials.
+      sys.env.get("GOOGLE_APPLICATION_CREDENTIALS") match {
+        case Some(json) =>
+          job.getConfiguration.set("fs.gs.auth.service.account.json.keyfile", json)
```

Comment:
Works locally with `export GOOGLE_APPLICATION_CREDENTIALS=x.json`. Is this sufficient though?